wdl-engine/src/backend/crankshaft.rs

```diff
@@ -458,4 +464,153 @@ impl TaskExecutionBackend for CrankshaftBackend {
             completed: completed_rx,
         })
     }
+
+    fn cleanup(&self, output_dir: &Path) -> Result<oneshot::Receiver<Result<()>>> {
+        let (tx, rx) = oneshot::channel();
+
+        if self.kind != CrankshaftBackendKind::Docker {
+            let _ = tx.send(Ok(()));
+            return Ok(rx);
+        }
+
+        let inner_backend = self.inner.clone();
+        let generator = self.generator.clone();
+        let output_path = output_dir.to_path_buf();
+
+        tokio::spawn(async move {
+            let result = async {
+                let (uid, gid) = get_uid_gid();
+                let ownership = format!("{uid}:{gid}");
+                info!(
+                    "cleanup target: '{}', attempting to set ownership to: {}",
+                    output_path.display(),
+                    ownership
+                );
+
+                let guest_out_dir = "/workflow_output";
+
+                if !output_path.exists() {
+                    info!("output directory does not exist, skipping cleanup");
+                    return Ok(());
+                }
+                if !output_path.is_dir() {
+                    bail!(
+                        "output directory `{path}` is not a directory",
+                        path = output_path.display()
+                    );
+                }
+
+                let output_mount = Input::builder()
+                    .path(guest_out_dir)
+                    .contents(Contents::Path(output_path.clone()))
+                    .ty(Type::Directory)
+                    // need write acces
+                    .read_only(false)
+                    .build();
+
+                let cleanup_task_name = format!(
+                    "wdl-engine-chown-cleanup-{}",
+                    generator
+                        .lock()
+                        .expect("generator should always acquire")
+                        .next()
+                        .expect("generator should never be exhausted")
+                );
+
+                let cleanup_resources = Resources::builder().cpu(0.1).ram(0.05).build();
+
+                let cleanup_task = Task::builder()
```

Comment:
Should this be defined by the inner backend? Can we think of any other backend-specific cleanup steps that would need to be run?

Reply:
Can do. But shouldn't we have as slim resources as possible for cleanup

Reply:
Ah, the comment is meant to be for the definition of the `cleanup_task`. The recursive `chown` is specific to Docker. My question was more general for the group as to whether we expect a necessary cleanup for other backends.

Reply:
We may one day want to add a "remove intermediate files" config option that on success of a workflow purges the filesystem of anything not saved as an `output`. That seems a long way off, but is in theory a use for this cleanup func?

Reply:
I think we can punt on this as my support for TES will actually break apart the Crankshaft backend into multiple `wdl-engine` (`docker`, `tes`, etc.) backends as they're different enough in terms of their behavior to require it.

Regarding its potential use to clean up "intermediate" files, I think we could build upon this to accept cleanup options that the backends can use to figure out what needs to be cleaned up (in the case of Docker, it can delete files in the local `work` directory; for TES that might mean making HTTP requests to delete cloud storage files, etc.).